pkg/cmd/subcommand/install.go

```diff
@@ -0,0 +1,506 @@
+		if exist := inventoryExist["Keda"]; exist == "" || i.WithUpgrade {
+			grp1.Go(func() error {
+				return i.installKeda(g1ctx, cl, operator)
+			})
+		} else {
+			fmt.Fprintln(w, ti.SkipTask("Keda"))
+		}
+	}
+
+	if i.WithKnative {
+		// If the Knative Serving component already exists in the cluster
+		// and the `--upgrade` parameter is not used, skip this step.
+		if exist := inventoryExist["Knative Serving"]; exist == "" || i.WithUpgrade {
+			grp1.Go(func() error {
+				return i.installKnativeServing(g1ctx, cl, operator)
+			})
+		} else {
+			fmt.Fprintln(w, ti.SkipTask("Knative Serving & Kourier"))
+		}
+	}
+
+	if i.WithShipWright {
+		// We must install Shipwright(and Tekton Pipelines)
+		// with or without the `--upgrade` parameter.
+		grp1.Go(func() error {
+			return i.installShipwright(g1ctx, cl, operator)
+		})
+	}
+
+	if i.WithCertManager && i.OpenFunctionVersion != "v0.3.1" {
+		// If the Cert Manager component already exists in the cluster
+		// and the `--upgrade` parameter is not used, skip this step.
+		if exist := inventoryExist["Cert Manager"]; exist == "" || i.WithUpgrade {
+			grp1.Go(func() error {
+				return i.installCertManager(g1ctx, cl, operator)
+			})
+		} else {
+			fmt.Fprintln(w, ti.SkipTask("Cert Manager"))
+		}
+	}
+
+	if i.WithIngress && i.OpenFunctionVersion != "v0.3.1" {
+		// If the Ingress Nginx component already exists in the cluster
+		// and the `--upgrade` parameter is not used, skip this step.
+		if exist := inventoryExist["Ingress Nginx"]; exist == "" || i.WithUpgrade {
+			grp1.Go(func() error {
+				return i.installIngress(g1ctx, cl, operator)
+			})
+		} else {
+			fmt.Fprintln(w, ti.SkipTask("Ingress Nginx"))
+		}
+	}
+
+	if err := grp1.Wait(); err != nil {
+		return errors.New(ti.TaskFail(err.Error()))
+	}
+
+	grp2, g2ctx := errgroup.WithContext(ctx)
+	defer g2ctx.Done()
+
+	grp2.Go(func() error {
+		return i.installOpenFunction(g2ctx, cl, operator)
+	})
+
+	if err := grp2.Wait(); err != nil {
+		return errors.New(ti.TaskFail(err.Error()))
+	}
+
+	end := time.Since(start)
+	fmt.Fprintln(w, ti.AllDone(end))
+	return nil
+}
+
+func (i *Install) checkConditionsAndGetInventory() map[string]string {
+	getVersionFromEnv := func(name string, defaultVersion string) string {
+		val, ok := os.LookupEnv(name)
+		if !ok {
+			return defaultVersion
+		} else {
+			return val
+		}
+	}
+
+	inventory := map[string]string{
+		"OpenFunction": i.OpenFunctionVersion,
+	}
+
+	// Update the corresponding conditions when WithAll is true
+	if i.WithAll {
+		i.WithDapr = true
+		i.WithKeda = true
+		i.WithIngress = true
+		i.WithKnative = true
+		i.WithShipWright = true
+		i.WithCertManager = true
+	}
+
+	// Update the corresponding conditions when WithAsyncRuntime is true
+	if i.WithAsyncRuntime {
+		i.WithDapr = true
+		i.WithKeda = true
+	}
+
+	// Update the corresponding conditions when WithSyncRuntime is true
+	if i.WithSyncRuntime {
+		i.WithKnative = true
```

Comment:
Can we use SyncRuntime for keda-http + deployment combination only, aka OpenFunctionSync runtime.
knative could be a separate option although it's now categorized into sync runtime too

Reply:
Understood, I've commented it for now